domains/certificates/Query.API/API/MeasurementsSyncer/MeasurementsSyncService.cs

```diff
@@ -3,39 +3,50 @@
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using API.ContractService.Clients;
 using API.MeasurementsSyncer.Metrics;
 using API.MeasurementsSyncer.Persistence;
 using DataContext.Models;
 using DataContext.ValueObjects;
 using MassTransit;
-using MeasurementEvents;
 using Measurements.V1;
 using Microsoft.Extensions.Logging;
+using HashedAttribute = API.ContractService.Clients.HashedAttribute;
+using Technology = DataContext.ValueObjects.Technology;
 
 namespace API.MeasurementsSyncer;
 
 public class MeasurementsSyncService
 {
+    public const string AssetId = "AssetId";
+    public const string TechCode = "TechCode";
+    public const string FuelCode = "FuelCode";
+
     private readonly ISlidingWindowState slidingWindowState;
     private readonly Measurements.V1.Measurements.MeasurementsClient measurementsClient;
-    private readonly IPublishEndpoint bus;
     private readonly SlidingWindowService slidingWindowService;
     private readonly IMeasurementSyncMetrics measurementSyncMetrics;
+    private readonly IStampClient stampClient;
     private readonly ILogger<MeasurementsSyncService> logger;
 
     public MeasurementsSyncService(ILogger<MeasurementsSyncService> logger, ISlidingWindowState slidingWindowState,
         Measurements.V1.Measurements.MeasurementsClient measurementsClient, IPublishEndpoint bus, SlidingWindowService slidingWindowService,
-        IMeasurementSyncMetrics measurementSyncMetrics)
+        IMeasurementSyncMetrics measurementSyncMetrics, IStampClient stampClient)
     {
         this.logger = logger;
         this.slidingWindowState = slidingWindowState;
         this.measurementsClient = measurementsClient;
-        this.bus = bus;
         this.slidingWindowService = slidingWindowService;
         this.measurementSyncMetrics = measurementSyncMetrics;
+        this.stampClient = stampClient;
     }
 
-    public async Task FetchAndPublishMeasurements(string meteringPointOwner, MeteringPointTimeSeriesSlidingWindow slidingWindow,
+    public async Task FetchAndPublishMeasurements(string meteringPointOwner,
```

Comment:
Can you take syncinfo as a parameter? instead of 5 of these parameters, so syncinfo and slidingwindow?

Reply:
done